ferrischat_webserver/src/users/bots/get_bots_by_user.rs

```diff
@@ -0,0 +1,55 @@
+use actix_web::{HttpRequest, HttpResponse, Responder};
+
+use ferrischat_common::types::{
+    BotsOwnedByUser, InternalServerErrorJson, NotFoundJson, User, UserFlags,
+};
+
+/// GET `/api/v0/users/{user_id}/bots`
+/// Get all bots owned by the user
+pub async fn get_bots_by_user(req: HttpRequest, auth: crate::Authorization) -> impl Responder {
+    let bigint_user_id = u128_to_bigdecimal!(auth.0);
+
+    let db = get_db_or_fail!();
+
+    let resp = sqlx::query!("SELECT * FROM bots WHERE owner_id = $1", bigint_user_id)
+        .fetch_all(db)
+        .await;
+
+    let bots = match resp {
+        Ok(resp) => resp
+            .iter()
+            .filter_map(|x| {
+                let resp = sqlx::query!("SELECT * FROM users WHERE id = $1", x.bot_id)
+                    .fetch_one(db)
+                    .await;
+
+                match resp {
+                    Ok(resp) => User {
+                        id: resp.id,
+                        name: resp.name,
+                        avatar: None,
+                        guilds: None,
+                        discriminator: resp.discriminator,
+                        flags: UserFlags::from_bits_truncate(resp.flags),
+                    },
+                    Err(e) => {
+                        return HttpResponse::InternalServerError().json(InternalServerErrorJson {
+                            reason: format!("database returned a error: {}", e),
+                            is_bug: false,
```

Comment:
should be `Option<false>`

Reply:
since the when

Reply:
since i made the struct

Reply:
![image](https://user-images.githubusercontent.com/64497526/142934907-e90df251-2593-4b0c-b2bf-aed9835de805.png)

Reply:
is not Option<bool>

Reply:
oh right. he changed it